endless/src/endless/EndlessUsbToolDlg.cpp

```diff
@@ -5080,6 +5081,19 @@ bool CEndlessUsbToolDlg::CreateUSBPartitionLayout(HANDLE hPhysical, DWORD &Bytes
 	IFFALSE_PRINTERROR(BytesPerSector * 24 == write_sectors(hPhysical, BytesPerSector, partitionStart[1], 24, zeroes), "Unable to clear the start of the ESP.");
 	free(zeroes);
 
+	// We've actually already called IOCTL_DISK_CREATE_DISK through rufus' InitializeDisk(), however that call
+	// doesn't set DiskId or MaxPartitionCount. There's a note somewhere in rufus that "If you don't call
+	// IOCTL_DISK_CREATE_DISK, the IOCTL_DISK_SET_DRIVE_LAYOUT_EX call will fail". It turns out that on windows 8.1
+	// and windows 7, in my testing, this is true. Windows 10 succeeds. The failure claims the disk is corrupt,
+	// GetLastError() code 1393. Perhaps this is because there is no GPT Disk Id set?
```

Comment:
I thought this error code sounded familiar. I think you have successfully debugged a bug first reported in November 2017. https://phabricator.endlessm.com/T19917